File: src/main/java/cz/forgottenempire/arma3servergui/installation/ServerInstallationDto.java
package cz.forgottenempire.arma3servergui.installation;

import cz.forgottenempire.arma3servergui.common.InstallationStatus;
import cz.forgottenempire.arma3servergui.common.ServerType;
import cz.forgottenempire.arma3servergui.steamcmd.ErrorStatus;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
class ServerInstallationDto {

    private ServerType type;
    private String version;
    private InstallationStatus installationStatus;
    private ErrorStatus errorStatus;
    private String lastUpdatedAt;
}

File: src/main/webapp/src/dtos/ServerInstallationDto.ts
import {ServerType} from "./ServerDto.ts";

export interface ServerInstallationsDto {
    serverInstallations: Array<ServerInstallationDto>
}

export interface ServerInstallationDto {
    type: ServerType,
    version: string,
    installationStatus: string,
    errorStatus: string,
    lastUpdatedAt: string
}
